SingularityService/src/main/java/com/hubspot/singularity/resources/S3LogResource.java

```diff
@@ -238,7 +238,7 @@ public SingularityS3Log call() throws Exception {
   }
 
   private List<SingularityS3Log> getS3Logs(Optional<String> group, Collection<String> prefixes) throws InterruptedException, ExecutionException, TimeoutException {
-    ListeningExecutorService executorService = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(Math.min(prefixes.size(), configuration.get().getMaxS3Threads()),
+    ListeningExecutorService executorService = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(Math.min(prefixes.size() == 0 ? 1 : prefixes.size(), configuration.get().getMaxS3Threads()),
```

Comment:
idk why i suggested that route -- can't we just short-circuit to `Collections.emptyList()` if there are no prefixes?